controllers/argocd/service.go

```diff
@@ -207,9 +220,17 @@ func (r *ReconcileArgoCD) reconcileRedisHAMasterService(cr *argoprojv1a1.ArgoCD)
 func (r *ReconcileArgoCD) reconcileRedisHAProxyService(cr *argoprojv1a1.ArgoCD) error {
 	svc := newServiceWithSuffix("redis-ha-haproxy", "redis", cr)
 	if argoutil.IsObjectFound(r.Client, cr.Namespace, svc.Name, svc) {
+		if ensureAutoTLSAnnotation(svc, common.ArgoCDRedisServerTLSSecretName, cr.Spec.Redis.WantsAutoTLS()) {
+			return r.Client.Update(context.TODO(), svc)
+		}
+		if !cr.Spec.HA.Enabled {
+			return r.Client.Delete(context.TODO(), svc)
+		}
```

Comment:
Shouldn't the check for HA enabled be before possibly updating auto TLS annotation (and then return)?